hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/JoinScanProcessorSupplier.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.impl.processor.TransformBatchedP;
+import com.hazelcast.jet.sql.impl.ExpressionUtil;
+import com.hazelcast.jet.sql.impl.JetJoinInfo;
+import com.hazelcast.jet.sql.impl.connector.keyvalue.KvRowProjector;
+import com.hazelcast.map.IMap;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.sql.impl.expression.Expression;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map.Entry;
+
+import static com.hazelcast.jet.Traversers.traverseIterable;
+import static com.hazelcast.jet.impl.util.Util.extendArray;
+
+@SuppressFBWarnings(
+        value = {"SE_BAD_FIELD", "SE_NO_SERIALVERSIONID"},
+        justification = "the class is never java-serialized"
+)
+final class JoinScanProcessorSupplier implements ProcessorSupplier, DataSerializable {
+
+    private static final int MAX_BATCH_SIZE = 1024;
```

Comment:
Why do we limit the batch size? We do `entrySet` repeatedly but only in order to have fresh data. But I think the batch can be unlimited here (and it's practically limited already, the inbox has limited size). It might even be better to scan for updated data once per second.

Reply:
Removed the limit. I would not cache `IMap` data as that would change the semantics - i.e. one could not update `IMap` then produce Kafka record and join it while currently this is the case for all the strategies we use internally.